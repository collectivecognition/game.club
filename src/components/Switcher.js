import { connect } from 'react-redux'
import React, { Component, PropTypes } from 'react'
import { Link } from 'react-router'

import { fetchGames } from '../actions'

require('./Switcher.styl')

export class Switcher extends Component {
  constructor(props) {
    super(props);

    this.searchTerm = ''

    this.onClick = this.onClick.bind(this);
    this.onChange = this.onChange.bind(this);
  }

  onClick() {
    this.props.dispatch(fetchGames(this.searchTerm));
  }

  onChange(event) {
    this.searchTerm = event.target.value;
  }

  render() {
    return (
      <div className="switcher">
        <input onChange={this.onChange}></input>
        <button onClick={this.onClick}>Search</button>
        {this.props.isFetching && 'Loading...'}
        <ul>
          {this.props.games.map((game, index) =>
            <li key={index}>
              <Link to={`/game/${game.id}`}>{game.name}</Link>
            </li>
          )}
        </ul>
      </div>
    )
  }
}

Switcher.PropTypes = {
  dispatch: PropTypes.func.isRequired
}

const mapStateToProps = (state, ownProps) => {
  return state.games;
}

export default connect(mapStateToProps)(Switcher);