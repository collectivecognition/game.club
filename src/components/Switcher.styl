.switcher
  padding: 20px
  background: #ddd