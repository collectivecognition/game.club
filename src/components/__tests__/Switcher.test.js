import React from 'react';
import render from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

import { Switcher } from '../Switcher';

jest.mock('api');

describe('Switcher', () => {
  it('should render initially', () => {
    const component = render.create(<Switcher games={[]}/>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should update searchTerm', () => {
    const component = mount(<Switcher games={[]}/>)
    const input = component.find('input')
    input.get(0).value = 'Zelda'
    input.simulate('change')
    expect(component.instance().searchTerm).toBe('Zelda')
  })

  it('should call onClick', () => {
    const component = mount(<Switcher games={[]} dispatch={jest.fn()}/>)
    const button = component.find('button')
    component.instance().onClick = jest.fn()
    button.click()
    expect(component.instance().onClick).toBeCalled()
  })
});