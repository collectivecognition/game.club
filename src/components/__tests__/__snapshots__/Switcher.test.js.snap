exports[`Switcher should render initially 1`] = `
<div
  className="switcher">
  <input
    onChange={[Function]} />
  <button
    onClick={[Function]}>
    Search
  </button>
  
  <ul />
</div>
`;
